features/flows/api/src/main/java/org/opennms/netmgt/flows/api/EnrichedFlow.java

```diff
@@ -0,0 +1,506 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2020 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2020 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.flows.api;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class EnrichedFlow {
+
+    public enum Locality {
+        PUBLIC,
+        PRIVATE
+    }
+
+    private Flow flow;
+
+    private String application;
+
+    private String convoKey;
```

Comment:
This is pure elasticsearch specific and can be calculated from other values.

Reply:
It would actually be nice to have this populated the same way that is done is Elasticsearch and avoid having to replicate that logic elsewhere.